pkgs/development/tools/clang-tools/default.nix

```
@@ -9,14 +9,14 @@ in stdenv.mkDerivation {
 
   dontUnpack = true;
 
+  cc = llvmPackages.stdenv.cc;
```

Would this not be more logical?

```suggestion
  inherit (stdenv) cc;
```

---

It would seem more logical, but it would not _work_ (because the default `stdenv.cc` is `gcc` and its `libcxx-cxxflags` are empty) :(.
Let us ignore the ill-considered last bullet in the commit message (sorry about that!) and let us try to think/type this through.

- `clangd`, at least as it is built in Nixpkgs, by accident or by design, does not know any include paths out of the box -- not even ones for the basic system libraries (`libc++` & `libc`) that the `llvm/clang` tool-chain it is a part of is configured to use.
- Perhaps it does not know those paths because they are not what `llvm/clang` would use by default?  Perhaps if `llvm/clang` were configured to use its own `libc++` & `libc` (is there one?), `clangd` would look at their respective include directories by itself?  I have no idea and that's not how things are, no reason to dwell on it then!
- The only reliable way, that I know of, to find out what the magic basic system library paths are for a given `cc` is to look under its `nix-support` directory.
- The  `llvm/clang` toolchains in Nixpkgs are configured to use `glibc` and the `libstdc++` that comes with the default `stdenv`'s `gcc`.
- Realistically, a user would use either `clang` or `gcc`.  Both are configured to use the same basic system libraries (barring possible complications with older versions, but if you use `clangd`, chances are you are not using it for a project built with `gcc` 6 or `clang` 3?  But who knows).
- Since `gcc/g++` does not need to be told where its  `libstdc++` is, its `nix-support/libcxx-cxxcflags` is empty.

The main problem here is, of course, that `nix-support/libc*-c*flags` files are there to tell the compiler where to look in addition to whatever it knows by itself -- and the `clangd` wrapper uses them to instead answer the question "where does the compiler look that we don't know?", so that's not exactly ideal.   In fact, it's a clear misuse.

So, is `cc = llvmPackages.stdenv.cc` the right _general_ solution?  Hell no.

Is it a working solution for whatever is assumed about a "normal" user of Nixpkgs who is using some recent version of Nixpkgs-packaged `gcc` or `llvm/clang` tool-chain and would like to also have a working index without too much hoop-jumping?  Looks so!

Is the whole encompassing tool-chain/wrapper design kind of incomplete?  Possibly, but I'm just trying to get the bloody thing to work for me...

I'll remove the ill-considered part of the commit message, thanks!